Python/import.c

```diff
@@ -1117,75 +1113,118 @@ list_frozen_module_names()
     return names;
 }
 
-static const struct _frozen *
-find_frozen(PyObject *modname)
+typedef enum {
+    FROZEN_OKAY,
+    FROZEN_BAD_NAME,    // The given module name wasn't valid.
+    FROZEN_NOT_FOUND,   // It wasn't in PyImport_FrozenModules.
+    FROZEN_DISABLED,    // -X frozen_modules=off (and not essential)
+    FROZEN_EXCLUDED,    // The PyImport_FrozenModules entry has NULL "code".
+    FROZEN_INVALID,     // The PyImport_FrozenModules entry is bogus.
```

Comment:
It means someone manually put invalid data into `PyImport_FrozenModules` (or that Tools/scripts/freeze_modules.py is broken).  We were already treating this case as `FROZEN_NOT_FOUND`.  I only teased out a separate enum member to make the condition more clear.

I don't mind adding more notes in the comment but to me the best thing to do is to look at the places where that status is emitted.

Reply:
> I don't mind adding more notes in the comment but to me the best thing to do is to look at the places where that status is emitted.

Do you mean in `set_frozen_error()`?

Reply:
I was referring to the enum value rather than the exception.  In the case of `FROZEN_INVALID`, that would be in `find_frozen()` where we return `FROZEN_INVALID` (or likewise `unmarshal_frozen_code()` or `_imp.get_frozen_object()`).

Reply:
https://bugs.python.org/issue45379